steel/src/host/builder.rs

```diff
@@ -230,11 +255,56 @@ impl<P> EvmEnvBuilder<P, EthBlockHeader, Url> {
     }
 }
 
+impl<P> EvmEnvBuilder<P, EthBlockHeader, History> {
```

Comment:
This type-state generic builder pattern is really nice!

Reply:
It is. Unfortunately, having non-compiletime config changes like
```rust
if let Some(beacon_api_url) = args.beacon_api_url { {
    builder = builder.beacon_api(beacon_api_url);
}
```
Is not possible with this pattern.